crates/re_renderer/shader/rectangle_fs.wgsl

```diff
@@ -14,78 +14,81 @@ fn tex_filter(pixel_coord: Vec2) -> u32 {
     }
 }
 
-fn decode_color(rgba_arg: Vec4) -> Vec4 {
-    var rgba = rgba_arg;
+fn normalize_range(sampled_value: Vec4) -> Vec4 {
+    let range = rect_info.range_min_max;
+    return (sampled_value - range.x) / (range.y - range.x);
+}
 
-    // Convert to linear space:
-    if rect_info.decode_srgb != 0u {
+fn decode_color(sampled_value: Vec4) -> Vec4 {
+    // Normalize the value first, otherwise premultiplying alpha and linear space conversion won't make sense.
+    var rgba = normalize_range(sampled_value);
+
+    // Convert to linear space. Skip if values are not in the 0..1 range (might be inf).
```

Comment:
We could use `ERROR_RGBA` for such values!

Reply:
good idea, now that the normalization happens always before, we can properly error out